src/classes/CRLP_RollupSoftCredit_TEST.cls

```diff
@@ -288,14 +288,16 @@ private class CRLP_RollupSoftCredit_TEST {
         insert opps;
 
         // Change the OCR Role on the org donation so that it will roll up as a soft credit.
-        OpportunityContactRole orgOCR = [SELECT Id, Role FROM OpportunityContactRole
+        OpportunityContactRole orgOCR = [SELECT Id, OpportunityId, ContactId, Role FROM OpportunityContactRole
                     WHERE Opportunity.AccountId = :orgAcct.Id AND IsPrimary = True LIMIT 1];
         orgOCR.Role = label.npo02.Household_Member_Contact_Role;
-        update orgOCR;
+        OpportunityContactRole duplicateSoftCreditOCR = orgOCR.clone();
```

Comment:
Soft Credit rollups will only roll up entries with `IsPrimary`=false, so cloning the queried row shouldn't have an impact here. It would be better to create a new OCR with the same Role, but making sure that `IsPrimary` is false. 

To really test this out across a bunch of records, I suggest doubling every non-primary OCR record using something along the lines of:

```
List<OpportunityContactRole> dupeOCRs = new List<OpportunityContactRole>();
for (OpportunityContactRole ocr : [SELECT OpportunityId, ContactId, Role FROM OpportunityContactRole WHERE IsPrimary=False AND Opportunity.IsWon=True]) {
   dupeOCRs.add(new OpportunityContactRole(
      OpportunityId = ocr.OpportunityId,
      ContactId = ocr.ContactId,
      Role = ocr.Role,
      IsPrimary = false));
}
insert dupeOCRs;
```

Reply:
@force2b, when I made the change by cloning the Primary OCR, the Soft Credit rollup amount double (Test failed). So I think the Rollup count Primary OCR as well. I will double check. But I get your point, I will create non-primary OCRs.

Reply:
Very interesting. Let's test a double IsPrimary OCR outisde of a unit test to see what happens.